presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -484,9 +485,10 @@ protected Type toPrestoType(int jdbcType, int columnSize, int decimalDigits)
                 return REAL;
             case Types.FLOAT:
             case Types.DOUBLE:
+                return DOUBLE;
             case Types.NUMERIC:
             case Types.DECIMAL:
-                return DOUBLE;
+                return createDecimalType(columnSize, decimalDigits);
```

Comment:
What about the case when we exceed precision/scale supported by Presto?
Maybe we should stay with `DOUBLE` then?

Reply:
🤔  ... As we support finite precision and no negative scale (e.g. no support for `decimal(40, -2)`), this is the only option. 👍

Reply:
I see other options. As for throw an error otherwise. Or make it configurable. I am a little concerned about braking backward compatibility here. But maybe this is not a huge problem.